### PR TITLE
VDR: v2 API to resolve service endpoints

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes https://github.com/nuts-foundation/nuts-node/issues/2868

Changed the solution to less convenience-function returning the endpoint, to a more flexible filter-style function that return a list of matching services. To aid the caller in parsing the service endpoint, I added the `endpointType` filter to control which service endpoint types (string, object, array) should be returned.

### CODE REVIEW

File: vdr/api/v2/api.go [hunk 1]
```diff
@@ -166,12 +168,58 @@ func (w Wrapper) UpdateService(ctx context.Context, request UpdateServiceRequest
 	return UpdateService200JSONResponse(*newService), nil
 }
 
-func (w Wrapper) AddVerificationMethod(ctx context.Context, request AddVerificationMethodRequestObject) (AddVerificationMethodResponseObject, error) {
+func (w *Wrapper) ResolveServiceEndpointByType(_ context.Context, request ResolveServiceEndpointByTypeRequestObject) (ResolveServiceEndpointByTypeResponseObject, error) {
```
Comment (gerardsn) [inline vdr/api/v2/api.go:171]: did spec allows multiple services with the same type, is this handled correctly everywhere (else)?

**Review by gerardsn — CHANGES_REQUESTED** [review]
double checking: this still allows compound services, the node just doesn't unpack it for you anymore?

File: vdr/api/v2/api.go (hunk 1 above)
Comment (reinkrul) [reply vdr/api/v2/api.go]: We now have a search/filter-style operation that can return multiple services.

**Review by gerardsn — APPROVED** [review]
LGTM. This needs some examples for users to understand possible endpointType formats. I'm fine adding them now, or when improving the API docs in general.

**Pull request merged**